cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -142,44 +99,32 @@ Slice::CsGenerator::getUnqualified(
     }
 }
 
-//
-// If the passed name is a scoped name, return the identical scoped name,
-// but with all components that are C# keywords replaced by
-// their "@"-prefixed version; otherwise, if the passed name is
-// not scoped, but a C# keyword, return the "@"-prefixed name;
-// otherwise, check if the name is one of the method names of baseTypes;
-// if so, prefix it with ice_; otherwise, return the name unchanged.
-//
 string
-Slice::CsGenerator::fixId(const string& name, unsigned int baseTypes, bool mangleCasts)
+Slice::CsGenerator::getMappedScoped(const ContainedPtr& p)
 {
-    if (name.empty())
-    {
-        return name;
-    }
-    if (name[0] != ':')
-    {
-        return lookupKwd(name, baseTypes, mangleCasts);
-    }
-    vector<string> ids = splitScopedName(name);
-    vector<string> newIds;
-    newIds.reserve(ids.size());
-    for (const auto& id : ids)
-    {
-        newIds.push_back(lookupKwd(id, baseTypes));
-    }
     stringstream result;
-    for (auto j = newIds.begin(); j != newIds.end(); ++j)
+
+    string name = p->mappedScoped();
```

Comment:
I find this code rather confusing: it operates on a `::`-separated name with the mapped C# name at the end.

Reply:
I removed this function.
Now you can tell the parser which scope-separator you want when you call `mappedScoped`.